### PR TITLE
Implement notifications job

### PR DESCRIPTION
**Pull request by aapomm**
This PR is a refactor and does not introduce any new features. The following changes are as follows:

- Decouple the notification broadcast from the ActivityTrackingJob and create a NotificationsBroadcastJob to handle the notification creation.
- Add actor argument for notification creation.

### CODE REVIEW

File: app/channels/notifications_channel.rb [hunk 1]
```diff
@@ -3,8 +3,8 @@ def subscribed
     stream_for current_user
   end
 
-  def check_unread
-    if current_user.notifications.unread.any?
+  def check_unread(data)
+    if current_user.notifications.unread.where(project_id: data['project_id']).any?
```
Comment (sean-yeoh) [inline app/channels/notifications_channel.rb:7]: I think this will result in an error since the notifications table doesn't have a `project_id` column

**Comment by sean-yeoh** [comment]
I find the overall flow kinda hard to follow:

1. After comment is created in the controller, call `broadcast_notifications` from `controllers/concerns/notified.rb`
2. `broadcast_notifications` calls `NotificationsBroadcastJob.perform_later` from `app/jobs/notifications_broadcast_job.rb`
3. `NotificationsBroadcastJob.perform_later` calls `Comment#notify` which subscribe the mentioned users and then notify subscribed and mentioned users.

Perhaps we can refactor `Comment#notify` to `NotificationsBroadcastJob` and use a `after_save` callback to subscribe mentioned users in `comment.rb`.

**Review by etdsoft — APPROVED** [review]
My biggest concern here is the separation of concerns between Activity and Notification: we're now tracking the "target" and the "actor" in both models when I'm not sure we should.

I'm happy for us to keep to the plan, but I think we should revisit before ending the cycle.

I'm not worried about migrations/existing notifications as it's unlikely users will care about a notification from 2 years ago.

File: app/jobs/notifications_broadcast_job.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class NotificationsBroadcastJob < ApplicationJob
+  queue_as :dradis_project
+
+  def perform(action:, actor_id:, notifiable_id:, notifiable_type:)
```
[the remark this answers is not in the document]
Comment (aapomm) [reply app/jobs/notifications_broadcast_job.rb]: It's always `current_user`. However, we lose access to this method once we leave the controller context which is why we pass it.
